src/transformers/generation/utils.py

```diff
@@ -2470,8 +2470,11 @@ def sample(
                         else (outputs.hidden_states,)
                     )
 
+            broken_batch_indices = torch.where(torch.max(next_token_scores, dim=-1).values == -torch.inf)[0]
```

Comment:
compute the batch indices where `next_token_scores` are all `-inf`

Reply:
Do we know how much this slows down generation? Checking every value for inf can slow down things quite a bit

Reply:
I haven't check this as @gante and @sgugger are not in favor of this change (they worry about the silent failure regarding the generation quality).

However, the check here is not `Checking every value for inf`. It compute max along each batch dim (which should be faster enough). If at certain batch indices we get `-inf`, it replace each of those batch index with an 1D array of `0`.
This should be faster than checking each places in a 2D tensor (I believe)